pkg/server/server.go

```diff
@@ -707,13 +708,26 @@ func NewServer(cfg Config, stopper *stop.Stopper) (serverctl.ServerStartupInterf
 	if rangeFeedBudgetFactory != nil {
 		nodeRegistry.AddMetricStruct(rangeFeedBudgetFactory.Metrics())
 	}
+
+	// Create an unlimited (for now) monitor for bytes used by raft entries. Use a
+	// 512 KiB allocation size, so that 96 raft scheduler workers would typically
+	// account as a 48 MiB baseline.
+	raftEntriesMetric := metric.NewGauge(metric.Metadata{
+		Name:        "raft.loaded_entries.bytes",
+		Help:        "Bytes allocated by raft Storage.Entries calls that are still kept in memory.",
+		Measurement: "Memory",
+		Unit:        metric.Unit_BYTES,
+	})
+	nodeRegistry.AddMetric(raftEntriesMetric)
+	raftEntriesMonitor := mon.NewMonitorWithLimit("raft-entries", mon.MemoryResource,
+		kvserver.RaftEntriesMemLimit, raftEntriesMetric, nil, /* maxHist */
+		512<<10 /* increment */, math.MaxInt64 /* noteworthy */, st)
+	raftEntriesMonitor.Start(ctx, nil /* pool */, mon.NewStandaloneBudget(math.MaxInt64))
```

Comment:
Should this inherit from `KVMemoryMonitor` or the KV memory budget or something? I know we don't really have a KV memory budget at the moment, but it seems like Raft processing is a pretty core part of KV memory usage.

Reply:
Hesitant to do this, since `KVMemoryMonitor` is inherited from SQL memory monitor, and I'm not sure about the implications of adding a new consumer to it. Maybe add a TODO to consolidate these various memory budgets?